src/sympc/protocol/falcon/falcon.py

```diff
@@ -404,3 +405,52 @@ def multiplication_protocol(
             x.ring_size,
         )
         return z_value
+
+    @staticmethod
+    def select_shares(x: MPCTensor, y: MPCTensor, b: MPCTensor) -> MPCTensor:
+        """Returns either x or y based on bit b.
+
+        Args:
+            x (MPCTensor): input tensor
+            y (MPCTensor): input tensor
+            b (MPCTensor): input tensor which is shares of a bit used as selector bit.
+
+        Returns:
+            z (MPCTensor):Returns x (if b==0) or y (if b==1).
+
+        Raises:
+            ValueError: If the selector bit tensor is not of ring size "2".
+        """
+        ring_size = int(b.share_ptrs[0].get_ring_size().get_copy())
+        shape = b.shape
+        if ring_size != 2:
+            raise ValueError(
+                f"Invalid {ring_size} for bit_injection,must be of ring size 2"
+            )
+        if shape is None:
+            raise ValueError("The selector bit tensor must have a valid shape.")
+        session = x.session
+
+        # TODO: Should be made to generate with CryptoProvider in Preprocessing stage.
+        c_ptrs: List[ReplicatedSharedTensor] = []
+        for session_ptr in session.session_ptrs:
+            c_ptrs.append(
+                session_ptr.prrs_generate_random_share(
+                    shape=shape, ring_size=str(ring_size)
+                )
+            )
+
+        c = MPCTensor(shares=c_ptrs, session=session, shape=shape)  # bit random share
+        c_r = ABY3.bit_injection(
+            c, session, session.ring_size
+        )  # bit random share in session ring.
+
+        tensor_type = get_type_from_ring(session.ring_size)
+        mask = (b + c).reconstruct(decode=False).type(tensor_type)
```

Comment:
Shouldn't this be XOR?

Reply:
Addition is implicitly xor for binary rings.

Reply:
We could also have it as xor, as xor implicitly call `add`

Reply:
Sure, I would change it, there would be more clarity.

Reply:
Done